### PR TITLE
[WIP TechDocs]: github workflow for techdocs container 

### LINKED ISSUE #2579: The techdocs-container on DockerHub uses the readme from the root of the Backstage repo instead of the package readme's. The techdocs-core plugin published to pypi uses no readme.
We should be more specific here and point to the readme for the actual package instead of the general backstage readme.

- [x] techdocs-core to publish [this README](https://github.com/spotify/backstage/blob/master/packages/techdocs-container/techdocs-core/README.md) to pypi. #2578 

- [ ] techdocs-container to publish [this README](https://github.com/spotify/backstage/blob/master/packages/techdocs-container/README.md) to Docker Hub. #2593

### PR DESCRIPTION
**Pull request by emmaindal**
This PR adds a workflow for publishing to Docker Hub mainly to be able to use image tags and specific [Docker Hub description.](https://github.com/docker/build-push-action#update-dockerhub-repo-description)

Still a draft PR because waiting for some response. Input on other solutions are more than welcome! 

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: .github/workflows/techdocs-container.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          push: true
+          tags: |
+            spotify/techdocs:latest
+            spotify/techdocs:1.0.0
+      # https://github.com/docker/build-push-action#update-dockerhub-repo-description
+      - name: Docker Hub Description
+        uses: peter-evans/dockerhub-description@v2
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
```
Comment (OrkoHunter) [inline .github/workflows/techdocs-container.yml:35]: From the docs [[1]](https://github.com/peter-evans/dockerhub-description) [[2]](https://github.com/docker/build-push-action), it looks like the password does not need to be the real password and can be overridden by an API token. So, let's rename :)

```suggestion
          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
```

Also, I'm not sure if it matters but the GitHub action examples uses `with` and the standalone example uses the `env`.

Comment (emmaindal) [reply .github/workflows/techdocs-container.yml:35]: @OrkoHunter I think it depends on if I need to generate on myself.  See "Note: Docker Hub Personal Access Tokens cannot be used as they are not supported by the API. See here and here for further details." [[1]](https://github.com/peter-evans/dockerhub-description#action-inputs)

Comment (emmaindal) [reply .github/workflows/techdocs-container.yml:35]: > Also, I'm not sure if it matters but the GitHub action examples uses `with` and the standalone example uses the `env`.

yeah I think I might need to set up these steps as well 

   ```
        name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
      -
        name: Login to DockerHub
        uses: docker/login-action@v1 
```

and the Login to DockerHub step uses `with` there.

Comment (OrkoHunter) [reply .github/workflows/techdocs-container.yml:35]: Oh no :'( I expected more from Docker. 

Thanks for correcting @emmaindal 🙇

**Comment by hooloovooo** [comment]
A more lightweight approach could be to just use the peter-evans/dockerhub-description@v2 action to update the description and leave the actual deployment to the existing webhook. Publishing seems easier to do without dockers workflow, but just using the docker CLI instead. https://medium.com/platformer-blog/lets-publish-a-docker-image-to-docker-hub-using-a-github-action-f0b17e5cceb3

**Comment by emmaindal** [comment]
I'll close this issue, the most important thing was tagging which you solved @OrkoHunter by the webhook. Let's figure out the description in another PR.

**Comment by OrkoHunter** [comment]
Sounds good to me @emmaindal

**Pull request closed without merging**